bundle/src/bundler.rs

```diff
@@ -123,28 +126,33 @@ impl BundlerUtil {
     }
 }
 
-/// Reads and decompresses a .tar.zstd file from an input stream into just a `meta.json` file
-///
-pub async fn parse_meta_from_tarball<R: AsyncBufRead>(input: R) -> anyhow::Result<VersionedBundle> {
+/// Reads and decompresses a .tar.zstd file from an input stream into just the specified file.
+/// This does not assume any specific order of files in the tarball, so it will read through all entries
+/// until it finds the specified file.
+async fn parse_file_from_tarball<R: AsyncBufRead>(
+    input: R,
+    file_name: &str,
+) -> anyhow::Result<Vec<u8>> {
```

Comment:
We're making some similar-ish code over in the ETL: #26797

A comment I made there that I think is relevant here:
https://github.com/trunk-io/trunk/pull/26797#discussion_r2125602708

tl;dr: If we are pulling and processing multiple files, we should split the operations that pull files and the operations for processing the files. Reason being that you don't want to re-read the tar multiple times potentially doubling the IO needed.

I can go deeper on how to execute this, but I figured you'd probably prefer to think through the solution.

Reply:
OK I pushed code that takes in a list of files and goes through the tarball once to extract them all (and stops early if it finds everything it needs) - please let me know if I missed the mark here